.github/workflows/test.yml

```
@@ -0,0 +1,31 @@
+name: Test
+on: [push, pull_request]
+
+jobs:
+  verify-docs:
+    name: Verify docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.15
+
+      - name: Install Terraform
+        uses: hashicorp/setup-terraform@v1
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+
+      - name: Generate docs
+        run: go generate
+
+      - name: Check if git detects no changes
+        run: |
+          if output=$(git status --porcelain) && [ -z "$output" ]; then
+            echo "Docs are up to date";
+          else
+            echo "Not all docs are up-to-date. Run 'go generate' and commit the changes.";
+            echo $output;
+            exit 1;
+          fi
```

I think this should do it:
```suggestion
      - name: Commit go generate if needed
        run: |
          git config user.name "GitHub Actions"
          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
          git add .
          git commit -m "Go generate"
          git push
```

---

Big ++ on this one :)

---

Cool idea! But I am hesitant with adding an automated commit without it being properly tested.

As it's not needed to reach the goal of this PR, I am going to merge the PR as-is for now. Feel free to open a separate PR for this addition.